dali/operators/generic/one_hot.cc

```diff
@@ -46,22 +49,45 @@ bool is_scalar(TensorShape<ndims> shape) {
   return volume(shape) == 1;
 }
 
-
-TensorShape<DynamicDimensions>
-determine_shape(TensorShape<DynamicDimensions> in_shape, int nclasses) {
-  if (is_scalar(in_shape)) {
-    return {nclasses};
+TensorShape<> determine_shape(TensorShape<> in_shape, int num_classes, int axis,
+                              int output_sample_dim) {
+  if (output_sample_dim == 1) {
+    return {num_classes};
   }
-  return shape_cat(in_shape, nclasses);
+  axis = axis < 0 ? in_shape.sample_dim() : axis;
+  int outer_axes = axis;
+  int inner_axes = in_shape.sample_dim() - axis;
+  auto outer =  in_shape.first(outer_axes);
+  auto inner = in_shape.last(inner_axes);
+  auto tmp = shape_cat(outer, num_classes);
+  return shape_cat(tmp, inner);
```

Comment:
```suggestion
  return shape_cat(shape_cat(outer, num_classes), inner);
```
The sequence `outer, num_classes, inner` is visible and it gets rid of `tmp` which doesn't say anything.

Reply:
Good idea, done.